src/test/java/mate/academy/booksstore/service/ShoppingCartServiceTest.java

```diff
@@ -0,0 +1,222 @@
+package mate.academy.booksstore.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import java.math.BigDecimal;
+import java.util.Optional;
+import java.util.Set;
+import mate.academy.booksstore.dto.CartItemDto;
+import mate.academy.booksstore.dto.CartItemRequestDto;
+import mate.academy.booksstore.dto.CartItemUpdateRequestDto;
+import mate.academy.booksstore.dto.ShoppingCartDto;
+import mate.academy.booksstore.exception.EntityNotFoundException;
+import mate.academy.booksstore.mapper.CartItemMapper;
+import mate.academy.booksstore.mapper.ShoppingCartMapper;
+import mate.academy.booksstore.model.Book;
+import mate.academy.booksstore.model.CartItem;
+import mate.academy.booksstore.model.Category;
+import mate.academy.booksstore.model.Role;
+import mate.academy.booksstore.model.RoleName;
+import mate.academy.booksstore.model.ShoppingCart;
+import mate.academy.booksstore.model.User;
+import mate.academy.booksstore.repository.BookRepository;
+import mate.academy.booksstore.repository.CartItemsRepository;
+import mate.academy.booksstore.repository.ShoppingCartRepository;
+import mate.academy.booksstore.repository.UserRepository;
+import mate.academy.booksstore.service.impl.ShoppingCartServiceImpl;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.testcontainers.shaded.org.apache.commons.lang3.builder.EqualsBuilder;
+
+@ExtendWith(MockitoExtension.class)
+class ShoppingCartServiceTest {
+    private static final Long EXIST_ID = 1L;
+
+    private static final Long INVALID_ID = Long.MAX_VALUE;
+
+    @Mock
+    private CartItemMapper cartItemMapper;
+
+    @Mock
+    private CartItemsRepository cartItemsRepository;
+
+    @Mock
+    private UserRepository userRepository;
+
+    @Mock
+    private BookRepository bookRepository;
+
+    @Mock
+    private ShoppingCartRepository shoppingCartRepository;
+
+    @Mock
+    private ShoppingCartMapper shoppingCartMapper;
+
+    @InjectMocks
+    private ShoppingCartServiceImpl shoppingCartService;
+
+    private ShoppingCart shoppingCart;
+
+    private ShoppingCartDto shoppingCartDto;
+
+    private CartItemRequestDto cartItemRequestDto;
+
+    private CartItem cartItem;
+
+    private CartItemDto cartItemDto;
+
+    private Book book;
+
+    private User user;
```

Comment:
Instantiate each of these inside the test method where you use it. 

```suggestion
    private ShoppingCart shoppingCart;

    private ShoppingCartDto shoppingCartDto;

    private CartItemRequestDto cartItemRequestDto;

    private CartItem cartItem;

    private CartItemDto cartItemDto;

    private Book book;

    private User user;
```